cql3/statements/select_statement.cc

```diff
@@ -379,6 +382,28 @@ select_statement::do_execute(service::storage_proxy& proxy,
     command->slice.options.set<query::partition_slice::option::allow_short_read>();
     auto timeout_duration = get_timeout(state.get_client_state(), options);
     auto timeout = db::timeout_clock::now() + timeout_duration;
+
+    if (aggregate && _selection->is_count()) {
+        query::forward_request req = {
+            .reduction_types = {query::forward_request::reduction_type::count},
+            .cmd = *command,
+            .pr = std::move(key_ranges),
+            .cl = options.get_consistency(),
+            .timeout = timeout,
+            .skip_metadata = true,
+        };
+
+        // dispatch execution of this statement to other nodes
+        return forwarder.dispatch(req).then([this] (query::forward_result res) {
+            auto meta = make_shared<metadata>(*_selection->get_result_metadata());
+            auto rs = std::make_unique<result_set>(std::move(meta));
+            rs->add_column_value(*res.query_results[0]);
+            return shared_ptr<cql_transport::messages::result_message>(
+                make_shared<cql_transport::messages::result_message::rows>(result(std::move(rs)))
+            );
+        });
+    }
+
```

Comment:
This is still in runtime, not prepare time.

Reply:
Do you want me to fold 12c32c87dc9a40ea2ffc3c7af315c64dfe5fcc9a into the 12c32c87dc9a40ea2ffc3c7af315c64dfe5fcc9a? The former moves it into prepare time.